server/jetstream_test.go

```diff
@@ -10772,54 +10772,44 @@ func TestJetStreamMirrorBasics(t *testing.T) {
 	nc, js := jsClientConnect(t, s)
 	defer nc.Close()
 
-	createStream := func(cfg *StreamConfig) *JSApiStreamCreateResponse {
-		t.Helper()
-		req, err := json.Marshal(cfg)
-		if err != nil {
-			t.Fatalf("Unexpected error: %v", err)
-		}
-		rm, err := nc.Request(fmt.Sprintf(JSApiStreamCreateT, cfg.Name), req, time.Second)
-		if err != nil {
-			t.Fatalf("Unexpected error: %v", err)
-		}
-		var resp JSApiStreamCreateResponse
-		if err := json.Unmarshal(rm.Data, &resp); err != nil {
-			t.Fatalf("Unexpected error: %v", err)
-		}
-		return &resp
+	createStream := func(cfg *nats.StreamConfig) (*nats.StreamInfo, error) {
+		return js.AddStream(cfg)
 	}
 
-	createStreamOk := func(cfg *StreamConfig) {
+	createStreamOk := func(cfg *nats.StreamConfig) {
 		t.Helper()
-		if scr := createStream(cfg); scr.Error != nil {
-			t.Fatalf("Expected error, got %+v", scr.Error)
+		if _, err := createStream(cfg); err != nil {
+			t.Fatalf("Expected error, got %+v", err)
 		}
 	}
 
 	// Test we get right config errors etc.
-	cfg := &StreamConfig{
+	cfg := &nats.StreamConfig{
 		Name:     "M1",
-		Storage:  FileStorage,
+		Storage:  nats.FileStorage,
 		Subjects: []string{"foo", "bar", "baz"},
-		Mirror:   &StreamSource{Name: "S1"},
+		Mirror:   &nats.StreamSource{Name: "S1"},
 	}
-
-	scr := createStream(cfg)
-	if scr.Error == nil || !strings.Contains(scr.Error.Description, "stream mirrors can not") {
-		t.Fatalf("Expected error, got %+v", scr.Error)
+	_, err := createStream(cfg)
+	if err == nil || !strings.Contains(err.Error(), "stream mirrors can not") {
+		t.Fatalf("Expected error, got %+v", err)
 	}
 
 	// Clear subjects.
 	cfg.Subjects = nil
 
 	// Source
-	scfg := &StreamConfig{
+	scfg := &nats.StreamConfig{
 		Name:     "S1",
-		Storage:  FileStorage,
+		Storage:  nats.FileStorage,
 		Subjects: []string{"foo", "bar", "baz"},
 	}
+
 	// Create source stream
 	createStreamOk(scfg)
+
+	time.Sleep(2 * time.Second)
```

Comment:
What is this for?

Reply:
Ah, oops. That's left over from draft. Removing.